redbot/core/bank.py

```diff
@@ -332,6 +332,26 @@ def _invalid_amount(amount: int) -> bool:
         await _conf.clear_all_members(guild)
 
 
+async def bank_local_clean(guild: discord.Guild, member: discord.member = None):
+    """Remove data for members who are no longer in the server"""
+    # FIXME: use-config-bulk-update
+    # We do not recommend that devs use this protected method as it is not safe to be used in
+    # all cased and this can end up corrupting your config file
+    group = _conf._get_base_group(_conf.MEMBER, str(guild.id))
+    if member is not None:
+        accounts = await group.all()
+        tmp = accounts.copy()
+    async with group.all() as member_data:
+        if member is not None:
```

Comment:
You have this backwards, the first have is `if member is not None`, so there is a `member`. However, you clean the entire server here. In the `else` where there is no `member`, you attempt to clean a specific member.